src/ngraph_encapsulate_op.cc

```diff
@@ -168,8 +169,21 @@ class NGraphEncapsulateOp : public OpKernel {
     }
 
     // Set the backend type for the op
-    OP_REQUIRES_OK(ctx,
-                   ctx->GetAttr<string>("ngraph_backend", &m_op_backend_name));
+    std::string backend_name;
+    OP_REQUIRES_OK(ctx, ctx->GetAttr<string>("ngraph_backend", &backend_name));
+    // Get the optional attributes
+    std::vector<std::string> optional_attributes =
+        BackendManager::GetOptionalAttributes(backend_name);
+    std::unordered_map<std::string, std::string> optional_attribute_map;
+    for (int i = 0; i < optional_attributes.size(); i++) {
+      std::string val;
+      // If an attribute does not exist, TF will return a non-ok status
+      OP_REQUIRES_OK(ctx, ctx->GetAttr<string>(optional_attributes[i], &val));
```

Comment:
These are optional attributes. So we should not be `OP_REQUIRES_OK`-ing here right?

Also do we have a unit test where we do not pass in optional attributes?

Reply:
Note: this is by design. Its designed to fail if an attribute is not present to prevent any accidental foot-shooting.

Reply:
The term `optional_parameter` i misleading as it's not really an optional!